adrs/3122196229/README.md

```diff
@@ -0,0 +1,55 @@
+# NodeJS files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-11-10
+* **Tags:** nodejs
+
+## Context
+
+Across the ecosystem, we can find all sort of files and directories naming
+conversions in NodeJS projects.
+
+That has caused misalignment, shifting the focus away from more important
+matters, also have created conventions that introduce problems such:
+
+* Disagreement around English language, making it even more difficult for the
+  non-English speaker to figure out the right decision.
+* Case sensitive vs. case insensitive file systems, causing issues in different
+  environments such as your local computer vs. CI/CD computers.
+* Format mismatching around separating English words.
+
+The `camelCase`, and/or `PascalCase` for naming convention, although this seems
+a good solution there are some caveats.
+
+What happen for the following cases `httpClient` vs. `HTTPClient` vs.
+`HttpClient`?
```

Comment:
Throwing away `camelCase` because there's no common agreement for edge cases like `HTTP` is throwing the baby out with the bath water.

Reply:
`HTTP` is just one of the many cases that existing across the CS dictionary.

Throw `camelCase` based on all the factors, no just about this one.

Reply:
Obviously both of our perspectives are subjective. 

If your goal with switching away from casing in names is to reduce cognitive load - I will be converting component names to `-` names MUCH more often then I will be hitting an edge case with an HTTP vs Http example.

Reply:
Since component (react) files suppose to use `param-case` as well, this shouldn't be the case, except when you have a legacy system, there is always a cost to some ADRs 🤷🏻 , hopefully, the next project is much better, and don't have to invest energy into this.

Reply:
I'm not swayed either way by this argument.

Even in a `kebab-case`/`snake_case` world, the casing of acronyms in `PascalCase` still needs to be solved.  


For example: 
```ts
// utils/http_client.ts
class HttpClient { 
  // ...
}
```

Reply:
Agreed that the problem still exists in the contents of the file but that is outside the scope of this ADR.

Reply:
What @tleef said